tests/gpflow/test_functions.py

```diff
@@ -371,3 +372,20 @@ def test_models_with_mean_functions_changes(model_class: Type[Any]) -> None:
     assert np.all(var_zero.numpy() == var_non_zero.numpy())
     # predictive mean changes after modifying mean function
     assert not np.all(mu_zero.numpy() == mu_non_zero.numpy())
+
+
+class test_bug_2091_ensureParameterTypeForLinearMeanFunction(unittest.TestCase):
```

Comment:
In GPflow we use pytest for testing rather than unittest. You can easily convert this by changing the name of the test class, and changing the self.assertRaises to pytest's equivalent (see comment below).

```suggestion
class TestIssue2091EnsureParameterTypeForLinearMeanFunction:
```

(Also, remember to remove the `unittest` import above.)